lib/src/http/common/util.dart

```diff
@@ -25,7 +32,11 @@ StreamTransformer decodeAttempt(Encoding encoding) {
       subscription = input.listen(
           (data) {
             try {
-              data = encoding.decode(data);
+              if (encode) {
```

Comment:
correct. Same note as above, this method is really only a utility for the two public API methods